wgpu/backends/wgpu_native/_helpers.py

```diff
@@ -228,55 +230,69 @@ def to_camel_case(name):
 
 
 class WgpuAwaitable:
-    """An object that can be waited for, either synchronously using sync_wait() or asynchronously using await.
+    """
+    Create an object representing the result of a wgpu call that requires a callback
+    to complete.  The code can then call "awaitable.wait_sync()" to wait for the result
+    synchronously, or "await awaitable.wait_async()" to perform an asynchronous wait.
 
-    The purpose of this class is to implememt the asynchronous methods in a
-    truely async manner, as well as to support a synchronous version of them.
+    The callback should call "awaitable.set_result()" when it has a result, or
+    "awaitable.set_error()" when it encounters an error.
     """
 
-    def __init__(self, title, result, callback, poll_function, finalizer, timeout=5.0):
+    def __init__(self, title, callback, finalizer, poll_function=None, timeout=5.0):
         self.title = title  # for context in error messages
-        self.result = result  # a dict that the callbacks writes to
         self.callback = callback  # only used to prevent it from being gc'd
-        self.poll_function = poll_function  # call this to poll wgpu
         self.finalizer = finalizer  # function to finish the result
-        self.maxtime = time.perf_counter() + float(timeout)
+        self.poll_function = poll_function  # call this to poll wgpu
+        self.timeout = timeout
+        self.event = asyncio.Event()
```

Comment:
There's a couple of problems with this implementation:

* It relies on `asyncio`, which basically closes the door for other aync frameworks. I really want to avoid that.
* AFAIK the event does not help, because it can only be set from the poll function.
* A sleep time of 0.1 s is a *realy* long time :)

Reply:
1. Now the code uses anyio, which can support trio and asyncio.  The only asynchronous call it makes is to sleep.
2. Yeah. There is a possibility that someone really is writing asynchronous code and that a different coroutine has called "poll".  But you're right, it's not really worth the complications.
3. Changed to 5ms.